be/src/common/status.h

```diff
@@ -287,7 +289,30 @@ inline const Status& to_status(const StatusOr<T>& st) {
 #define AS_STRING_INTERNAL(x) #x
 #endif
 
-// Some generally useful macros.
+#if defined(ENABLE_STATUS_FAILED)
+#define RETURN_INJECT()                                    \
+    std::stringstream ss;                                  \
+    ss << "INJECT ERROR: " << __FILE__ << " " << __LINE__; \
+    return Status::Unknown(ss.str());
+
+#define RETURN_IF_WITH_INJECT(stmt)                                                                       \
+    do {                                                                                                  \
+        uint32_t seed = GetCurrentTimeNanos();                                                            \
+        seed = ::rand_r(&seed);                                                                           \
+        uint32_t boundary_value = RAND_MAX / (1.0 * config::probability_of_inject);                       \
+        if (seed <= boundary_value) {                                                                     \
+            RETURN_INJECT();                                                                              \
+        } else {                                                                                          \
+            auto&& status__ = (stmt);                                                                     \
+            if (UNLIKELY(!status__.ok())) {                                                               \
+                return to_status(status__).clone_and_append_context(__FILE__, __LINE__, AS_STRING(stmt)); \
+            }                                                                                             \
+        }                                                                                                 \
+    } while (false)
+#else
+#define RETURN_IF_WITH_INJECT(stmt) RETURN_IF_ERROR(stmt)
+#endif
```

Comment:
Maybe we could make `RETURN_IF_ERROR` calls `RETURN_IF_WITH_INJECT` oppositely to avoid modify too many codes and hide for the caller whether it is injected?

Reply:
We can't replace every use of RETURN_IF_ERROR with RETURN_IF_WITH_INJECT due to factors:
- starrocks_main.cpp: replace this range will cause failed to start.
- others(linclude storage directory, features like compactions/Import...): this range is not our tearget range.

Reply:
Could we add two configs to decide whether include or exclude the specific modules? like [--vmodule in VLOG](https://github.com/google/glog#verbose-logging).